### DISCUSSION TITLE
Using ModuleInitializer within source generator does not work

### DISCUSSION THREAD

**Discussion by Cheatoid** [dotnet/runtime#49048] [Q&A] [answered]
As the title says.
I've tried to add a simple module initializer method via source generator, but that doesn't seem to be working.
Any ideas why? Is this another limitation? Or a bug?

**Comment by Cheatoid** [comment, marked as answer]
Well this one is strange, but I just figured what is preventing it from working.
After several hours of trial and error...
In my source generator csproj I had:
```xml
<PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.10.0-1.final" PrivateAssets="all"/>
```
For whatever reason that makes it not working.
I tried out some different (older) version:
```xml
<PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.8.0" PrivateAssets="all"/>
```
And bam, it just magically works now. I am very much confused as to why this is...
To confirm it, I switched back to the newer version, and guess what -- it doesn't work again.
So it is literally something about the newer version which makes it not working.
Looking at the output window I saw:
> Could not load file or assembly 'Microsoft.CodeAnalysis, Version=3.10.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.

I would like to hear an explanation for this.

**Reply by alexrp** [reply]
Most IDEs/SDKs are using Roslyn 3.8.0 or 3.9.0. So if you compile an analyzer/generator against a newer Roslyn version, it will fail to load at design/build time.

Note that the Roslyn assemblies you reference in your generator project are just used to build it; they are not the ones that are actually loaded in memory while the compiler/IDE is running.
